src/main/python/city_specific_analysis/nyc_analysis.py

```diff
@@ -0,0 +1,1146 @@
+"""
+this is compilation of functions to analyse BEAM-related data for NYC simulation
+"""
+
+from urllib.error import HTTPError
+
+import matplotlib.pyplot as plt
+import numpy as np
+import time
+import datetime as dt
+import pandas as pd
+
+from shapely.geometry import Point
+from shapely.geometry.polygon import Polygon
+from io import StringIO
+
+
+def get_output_path_from_s3_url(s3_url):
+    """
+    transform s3 output path (from beam runs spreadsheet) into path to s3 output
+    that may be used as part of path to the file.
+
+    s3path = get_output_path_from_s3_url(s3url)
+    beam_log_path = s3path + '/beamLog.out'
+    """
+
+    return s3_url \
+        .strip() \
+        .replace("s3.us-east-2.amazonaws.com/beam-outputs/index.html#", "beam-outputs.s3.amazonaws.com/")
```

Comment:
Should this go into a timestamped bucket? Otherwise it is going to constantly overwrite the same file, correct?

Reply:
This is fine, the code is just to transform the output path from "BEAM Deploy Status and Run Data" spreadsheet into a path usable in python scripts.

to transform this
https://s3.us-east-2.amazonaws.com/beam-outputs/index.html#output/sfbay/sfbay-smart-base__2021-03-02_15-38-09_vtl
into this
https://beam-outputs.s3.amazonaws.com/output/sfbay/sfbay-smart-base__2021-03-02_15-38-09_vtl